src/lib/tls/tls13/tls_preshared_key.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * TLS 1.3 Preshared Key Container
+ * (C) 2023 Jack Lloyd
+ *     2023 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#ifndef BOTAN_TLS_PRESHARED_KEY_H_
+#define BOTAN_TLS_PRESHARED_KEY_H_
+
+#include <botan/secmem.h>
+#include <botan/strong_type.h>
+
+#include <vector>
+
+namespace Botan::TLS {
+
+/// @brief holds a PSK identity as used in TLS 1.3
+using PresharedKeyID = Strong<std::vector<uint8_t>, struct PresharedKeyID_>;
```

Comment:
Maybe use `std::string` for the external PSK identity? Typically, this is a human-readable label for a PSK. Maybe take inspiration from other TLS implementations.